plugins/performance-lab/includes/admin/local-plugin-fallback.php

```diff
@@ -0,0 +1,225 @@
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_file Plugin file path.
+	 * @return string Plugin description from readme or empty string.
+	 */
+	function perflab_get_plugin_readme_description( string $plugin_file ): string {
+		if ( ! defined( 'WP_PLUGIN_DIR' ) ) {
+			return '';
+		}
+
+		$plugin_dir   = dirname( WP_PLUGIN_DIR . '/' . $plugin_file );
+		$readme_files = array( 'readme.txt', 'README.txt', 'readme.md', 'README.md' );
+
+		foreach ( $readme_files as $readme_file ) {
+			$readme_path = $plugin_dir . '/' . $readme_file;
+			if ( file_exists( $readme_path ) ) {
+				$readme_content = file_get_contents( $readme_path );
+				if ( false !== $readme_content ) {
+					// Parse description from readme - look for description after "== Description ==".
+					if ( 1 === preg_match( '/==\s*Description\s*==(.*?)(?==|\z)/is', $readme_content, $matches ) ) {
+						$description = trim( $matches[1] );
+						// Remove markdown formatting and clean up.
+						$description = preg_replace( '/\*\*(.*?)\*\*/', '$1', $description ) ?? $description;
+						$description = preg_replace( '/\*(.*?)\*/', '$1', $description ) ?? $description;
+						$description = preg_replace( '/\[([^\]]+)\]\([^\)]+\)/', '$1', $description ) ?? $description;
+						return trim( $description );
+					}
+				}
+			}
+		}
+
+		return '';
+	}
+
+	/**
+	 * Sanitizes plugin description for display.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $description Raw plugin description.
+	 * @return string Sanitized description.
+	 */
+	function perflab_sanitize_plugin_description( string $description ): string {
+		if ( '' === $description ) {
+			return '';
+		}
+
+		// Strip all HTML tags and decode entities.
+		$description = wp_strip_all_tags( $description );
+		$description = html_entity_decode( $description, ENT_QUOTES, 'UTF-8' );
+
+		return trim( $description );
+	}
+
+	/**
+	 * Parses the requires_plugins from plugin headers.
+	 *
+	 * This attempts to extract required plugins from various possible header formats.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array<string, string> $plugin_headers Plugin headers array.
+	 * @param string                $plugin_slug    Plugin slug to determine dependencies.
+	 * @return string[] Array of required plugin slugs.
+	 */
+	function perflab_parse_requires_plugins( array $plugin_headers, string $plugin_slug ): array {
+		$requires_plugins = array();
+
+		// Check for RequiresPlugins header (WordPress 6.5+).
+		if ( isset( $plugin_headers['RequiresPlugins'] ) && '' !== $plugin_headers['RequiresPlugins'] ) {
+			$plugins          = array_map( 'trim', explode( ',', $plugin_headers['RequiresPlugins'] ) );
+			$requires_plugins = array_merge( $requires_plugins, $plugins );
+		}
+
+		// For known Performance Lab plugins, add their specific dependencies.
+		// Embed Optimizer has a hard dependency on Optimization Detective.
+		if ( 'embed-optimizer' === $plugin_slug ) {
+			$requires_plugins[] = 'optimization-detective';
+		}
```

Comment:
This logic is now duplicated with:

https://github.com/WordPress/performance/blob/4f17dd77eede251e98fa7c95e73f7087ff0254c4/plugins/performance-lab/includes/admin/plugins.php#L429-L432

It would be good to extract this into a separate helper function. 

In fact, this information could be added as part of `perflab_get_standalone_plugin_data()`, like as a new `suggests_plugins` key.